### ISSUE TITLE
QuickSetting long press action

### ISSUE THREAD

**Issue by ignoramous** [celzero/rethink-app#402]
Some apps show a menu on long press on the QuickSetting tile, while others open Settings page of their app.

If we are showing menu, then may be showing options for different modes (DNS, DNS + Firewall, Pause) is an option. If not, taking user to app's homescreen is another.

It would be nice to customize since the default long press action (on Android 11 at least) opens up Android Settings page instead.

**Comment by ignoramous** [comment]
Apparently, Tasker can call into [these shortcuts](https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts):

https://github.com/tailscale/tailscale/issues/3547#issuecomment-1191413818

**Comment by ignoramous** [comment]
See also: #774

**Comment by ignoramous** [comment]
Yes, upon looking into it further app 'shortcuts' would be the correct way not app 'activities'. And yes that would work with macrodroid.

From my understanding app shortcuts will work fine as long as they are not 'deep shortcuts' (generally meaning dynamic and pinned shortcuts). This [post](https://www.reddit.com/r/tasker/comments/gkounp/pluginbeta_taskerlaunchershortcut/) on reddit goes into depth about how these 'deep shortcuts' can't be launched by other apps other than default launcher apps or currently active voice interaction service. If I understand correctly, this means that static shortcuts are the type needed for apps such as tasker and macrodroid to use them.

Don't know if that clarification was already known or not but thought I might mention it anyways.

_Originally posted by @Crunchbits in https://github.com/celzero/rethink-app/issues/969#issuecomment-1679950626_

**Comment by JHBalaji** [comment]
Building on this request, I needed to expose them as dynamic shortcuts for use with Samsung Routines, so I implemented a standalone controller. Here is it [as apk as](https://github.com/JHBalaji/rethinkdns-dynamic-shortcuts/releases/tag/v1) well as source.
